zookeeper-server/src/main/java/org/apache/zookeeper/server/quorum/QuorumPeer.java

```diff
@@ -416,6 +416,22 @@ public void checkAddressDuplicate(QuorumServer s) throws BadArgumentsException {
         LOOKING, FOLLOWING, LEADING, OBSERVING;
     }
 
+    /**
+     * (Used for monitoring) When peer is in non-LOOKING phase, this shows the
+     * current phase of Zab protocol that peer is running.
+     */
+    public enum ZabState {
+        NONE, DISCOVERY, SYNCHRONIZATION, BROADCAST;
```

Comment:
Do we need 'NONE' here? As the comment said, only ZAB is only relevant for non-looking phase, but later in the code we are setting the state to 'NONE' for non looking peer. I guess we could:
* Update comment saying 'NONE' maps to the looking phase
Or 
* Remove 'NONE' from the state (which might be more consistent with traditional ZAB literatures), and don't set the zab state for looking peers.

Reply:
Most of the Zab literature talks about a three phases of Zab:
1. discovery
2. synchronization
3. broadcast

Left implicit is the degenerate phase:
0. maybe in leader election

(see https://distributedalgorithm.wordpress.com/2015/06/20/architecture-of-zab-zookeeper-atomic-broadcast-protocol/ or http://www.tcs.hut.fi/Studies/T-79.5001/reports/2012-deSouzaMedeiros.pdf)

I'd say that we do want a concept of the degenerate state. NONE is a reasonable thing to call it, as is ELECTION to my mind. I'll modify the comment to remove the mention of "nonlooking".

Reply:
`ELECTION` sounds good to me.